browser/javascript/onloadHelpers/generateCalculator.js

```diff
@@ -0,0 +1,35 @@
+function generateCalculator(name, windowName){
+  function attachCalculatorListeners(calculatorName, visitorId, calculatorHistory = []){
+    var calculator = new Calculator(0, 0, 0, null, false, visitorId, calculatorName, calculatorHistory)
+
+    numberButtons.forEach(function(button){ addNumberListener(button, calculator) })
+    operationButtons.forEach(function(button){ addOperationListener(button, calculator) })
+    addEqualsListener(equalsButton, calculator)
+    addAllClearListener(allClearButton, calculator)
+    addNegativeListener(negativeButton, calculator)
+    addDecimalListener(decimalButton, calculator)
+    addKeyDownListener(calculator)
+    addActiveCalculatorListener(calculator, true)
+    showNumberOnScreen('previous', calculator)
+  }
+
+  var numberButtons = Array.from(document.querySelectorAll(`div#${name} button.calculator-button-number`))
+  var operationButtons = Array.from(document.querySelectorAll(`div#${name} button.calculator-button-operation`))
+  var equalsButton = Array.from(document.querySelectorAll(`div#${name} button.calculator-button-equals`))[0]
+  var allClearButton = Array.from(document.querySelectorAll(`div#${name} button.calculator-button-ac`))[0]
+  var negativeButton = Array.from(document.querySelectorAll(`div#${name} button.calculator-button-neg`))[0]
+  var decimalButton = Array.from(document.querySelectorAll(`div#${name} button.calculator-button-decimal`))[0]
+
```

Comment:
I think there might be a way to dry up the code in line 16 to 21. Is there a way you can make this into another function where you pass in the class name as an argument as well?